doc/helplink.txt

```diff
@@ -17,13 +17,20 @@ http://meta.vi.stackexchange.com/a/1250/51
 ==============================================================================
 COMMANDS                                                   *helplink-commands*
 
-:Helplink {format}                                                 *:Helplink*
-
-        Try to find the tag closest to the cursor (backwards search only) and
-        format it. If multiple tags are found on the same line, it asks the
-        user which one to pick unless the cursor is currently on one of the
-        tags, in which case we assume the user wants to copy that tag and
-        don't ask for confirmation.
+:Helplink [subject] [format]                                       *:Helplink*
```

Comment:
I'm not sure if this command signature is still accurate? I had some trouble following the code to be honest, so I did some tests and inside a help buffer `:Helplink html` still works as before, whereas outside of it it's not `:Helplink :ls html`.

I find it a bit confusing, since the parameters now depend on the context of which buffer you have open.

Maybe it's better to have two commands for this, instead of "overloading" this one? That would also solve the other issue in my previous comment about `:Helplink` being globally available, and it would make the code easier to follow.

Or we could have `:Helplink` and `:Helplink!`? Not sure.

What do you think?

Reply:
Yeah it is confusing that the command works differently in help buffers to non-help buffers. I tried a few variation but didn't come up with anything that felt more natural to me.

I tried to make the command as forgiving as possible, so the order of arguments doesn't matter - `:Helplink :ls html` and `:Helplink html :ls` both work. The downside to this is that, as you say, the `helplink#link()` function becomes a bit messy. Especially since it also has to check whether it's being called from a `help` buffer.

Maybe a better approach is to split the function into two versions - leave the original as it was, and then add a new one with a positional `subject` argument and an optional `format` argument, i.e.:

```vim
fun! helplink#linkglobal(subject, ...) abort
```

Then the original code to create the `:Helplink` command could be restored for `help` buffers.

A new autocommand could then be added to create the global `:Helplink` command for non-`help` buffers, but only if the user has set `g:helplink_global_command`.

How does that all sound to you?